### ISSUE TITLE
Deepseek Sparse Attention Auxiliary Loss Kernel Formulation

### ISSUE THREAD

**Issue by floatingtrees** [meta-pytorch/attention-gym#476]
Indexer loss is expressed as the KL divergence between `attention_probs` and `indexer_probs`, where `attention_probs[batch, i, j] = probs[batch, k, i, j].sum(dim=1) / probs[batch, k, i, j].sum(dim=(1, -1))`, where `probs = softmax(Q_selected @ cat(K_selected, sliding_window_branch))[selected_indices]`. 

KL divergence is expressed as follows: 
$D_{\mathrm{KL}}(P \|\| Q)= \sum_i P(i)\log\left(\frac{P(i)}{Q(i)}\right)$

This means that to compute loss in one pass, we need to know the values of `indexer_attention_probs[i, j]` in `attention_probs[i, j]` at the same time, so `sparse_kv_lse` and `indexer_lse` will both be required.

My idea for the kernel is that each CTA computes a single element of a `(b, query_length)` tensor, which is then reduced into loss. 
For notation: 
Let `Q` be a `(num_heads, head_dim)` shaped tile from `query[batch, :, query_index, :]`, `K_i` be a `(chunk_length, head_dim)` shaped tile from `key[batch, :, i * chunk_length: (i+1) * chunk_length]`, `idx_Q` be a `(num_heads, idx_head_dim)` shaped tile from `index_query[batch, :, query_index, :]`, `idx_K_i` be a `(chunk_length, index_head_dim)` shaped tile from `key[batch, :, i * chunk_length: (i+1) * chunk_length]`

The steps roughly are: 
1. For i in range(key_length / chunk_length)
a. Compute `Q @ K_i`, `idx_Q @ idx_K_i`
b. Exponentiate and normalize `Q @ K_i` by `full_lse` to get `P'_i`, in shape `(num_heads, chunk_length)`
c. Exponentiate and normalize `idx_Q @ idx_K_i` by `index_lse` to get `idx_P_i`, in shape `(1, chunk_length)`
d. Reduce `P'_i` across heads, normalize by `sparse_kv_lse` to get `P_i`, in shape of `(1, chunk_length)`
e. Compute KL divergence with these probs and add into an accumulator
